### PR TITLE
Improve pmp interface

### PR DESCRIPTION
**Pull request by francois141**
This new interface allow to interact with the pmp module in a more friendly way. It avoids confusion such as forgetting dividing the values by 4 in the pmp entries. Overall it should make the pmp code easier to read and debug.

### CODE REVIEW

File: src/arch/pmp.rs [hunk 1]
```diff
@@ -219,8 +225,28 @@ impl PmpGroup {
...
+        assert!(
+            permissions < 8,
+            "Permissions should not set NAPOT or TOP bits"
+        );
+        self.set(idx, build_tor(until), permissions | TOR);
+    }
+
+    pub fn set_inactive(&mut self, idx: usize, addr: usize) {
+        self.set(idx, build_tor(addr), INACTIVE);
+    }
```
Comment (CharlyCst) [inline src/arch/pmp.rs:246]: Those functions are cool, but they lack some good documentation :)
They will be used in a bunch of places, so having a good doc comment would be great. Can you add an explanation for each of those?

**Review by CharlyCst — COMMENTED** [review]
Those changes are super cool, thanks for working on those :) 

Could you just add some doc comments for the new functions to set PMP entries? Thanks!

File: src/arch/pmp.rs (hunk 1 above)
Comment (francois141) [reply src/arch/pmp.rs:246]: Done!

**Comment by francois141** [comment]
@CharlyCst Ready to review again!

**Review by CharlyCst — APPROVED** [review]
Thanks, I'll rebase and merge :)

**Pull request merged**